src/Adapter/Product/CommandHandler/AddProductToPackHandler.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\CommandHandler;
+
+use Pack;
+use PrestaShop\PrestaShop\Adapter\Product\AbstractProductHandler;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\ValueObject\CombinationId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Command\AddProductToPackCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\CommandHandler\AddProductToPackHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductPackingException;
+use PrestaShopException;
+
+final class AddProductToPackHandler extends AbstractProductHandler implements AddProductToPackHandlerInterface
```

Comment:
I'm not sure any more about what we decided This is related to your comment here https://github.com/PrestaShop/docs/pull/558/files#r442763264

At first we talked about the trio `AddProductToPackCommand` `RemoveProductFromPackCommand` (and additional one `UpdatePackedProductQuantityCommand` as you said)

But I believe after discussion with @matks we decided that it would be simpler to have a single `UpdateProductPackCommand` which contains all the items of the pack (along with their quantities I guess)

It also matches the note from the PR https://github.com/PrestaShop/docs/pull/558/files#diff-c85d131d0c114e4a1d61d32b2f9cccfbR70

Reply:
Ok, Ill check what I can do